### ISSUE TITLE
Running against test data, leads to early stopping and very low F1 score

### ISSUE THREAD

**Issue by mahmoudyusuf94** [mohamedgabr96/NeuralDialectDetector#2]
I am trying to run your model against test data (which I got from NADI), I put the test data alongside dev and train data. I just changed the test loader/generator to read from the proper file:

In datset_util.py -> parse_and_generate_loaders

`
   test_generator = parse_and_generate_loader(path_to_data_folder, tokenizer, params_dev, classes_list, split_set="test", locale="ar", masking_percentage=masking_percentage, class_to_filter=class_to_filter, regional_mapping=regional_mapping_content, filter_w_indexes=filter_w_indexes, pred_class=pred_class, max_seq_len=max_seq_len, is_province=is_province, is_MSA=is_MSA, handle_imbalance_sampler=sampler_imbalance)

`

and running against the following config:

```
adam_epsilon: 1.0e-08
adapter_type: plain_adapter
batch_size: 32
bottleneck_dim: 384
checkpoint_on_improvement: true
checkpointing_freq: 1
checkpointing_on: false
checkpointing_path: checkpoints_marbert
class_index: -1
classif_dropout_rate: 0.3
current_adapter_to_train: 6
device: cuda
early_stopping: true
early_stopping_patience: 10
handle_imbalance_sampler: false
improvement_check_freq: 100
indexes_filtration_path: null
initial_learning_rate: 5.0e-05
is_MSA: false
is_province: false
lr-mini-epoch-size: 50
inv-sqrt-lr-temperature: 2
inv-sqrt-lr-min-factor: 0.001
inv-sqrt-lr-max-factor: 1
masking_percentage: 0
max_ex_per_class: null
max_sequence_length: 100
model_class: ArabicDialectBERT
model_name_path: UBC-NLP/MARBERT
neptune_experiment_name: MARBERT_DA_Country_100
neptuneaiAPI: eyJhcGlfYWRkcmVzcyI6Imh0dHBzOi8vdWkubmVwdHVuZS5haSIsImFwaV91cmwiOiJodHRwczovL3VpLm5lcHR1bmUuYWkiLCJhcGlfa2V5IjoiYzNlOTI4MDUtMWEwMC00N2U3LWEzYTAtZjAyMzA2YzE1YjI3In0=
no_total_adapters: 21
num_epochs: 5
one_class_filtration: null
path_to_data: NADI2021_DEV.1.0/Subtask_1.2+2.2_DA
run_title: MARBERT_DA_Country_100
save_final_model: false
seed_list:
- 1234
- 611
- 520
- 300
- 760
- 42
- 1111
- 1234
- 10
- 1
- 20
- 5
- 2
- 100
- 512
stage_2_training: true
use_adapt_after_fusion: true
use_adapters: true
use_neptune: false
use_regional_mapping: false
use_vert_att: false
vatt-bottleneck_dim: 384
vatt-final-adapter: false
vatt-positional-keys: 'sinosoid'
# vatt-positional-keys: 'random'
vatt-use-common-transform: false
warmup_steps: 250
```

which is the default config in your repo with some modifications:
use_neptune: false
num_epochs: 5
inv-sqrt-lr-max-factor: 1

The issue is the model seems to be overfitting, it applies early stopping and stops after the 4th epoch, and got horrible test results:

`{"seed_0": {"DEV": {"F1": 0.3056884710732168, "Accuracy": 0.5087999701499939, "Loss": 2.029092396600559}, "TEST": {"F1": 0.04380548432636026, "Accuracy": 0.10939999669790268, "Loss": 0.0}}, "Agg": {"TRAIN": {"Accuracy": [], "Loss": []}, "DEV": {"Accuracy": 0.5087999701499939, "Loss": 2.029092396600559}, "TEST": {"Accuracy": 0.10939999669790268, "Loss": 0.0}}}`

While if I don't use test data and leave it against dev data only, I can train for as many epochs I want.

I don't know what is the problem? and how does introducing test data affect early stopping? I understand it should be related to dev only?
